openquake/hazardlib/gsim/projects/acme_base.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2012-2021 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+from copy import deepcopy
+
+from openquake.hazardlib.gsim.base import CoeffsTable
+
+from openquake.hazardlib.gsim.nga_east import _at_percentile
+
+class CoeffsTableACME(CoeffsTable):
+
+    def __getitem__(self, imt):
+        """
+        Return a dictionary of coefficients corresponding to ``imt``
+        from this table (if there is a line for requested IMT in it),
+        or the dictionary of interpolated coefficients, if ``imt`` is
+        of type :class:`~openquake.hazardlib.imt.SA` and interpolation
+        is possible.
+
+        :raises KeyError:
+            If ``imt`` is not available in the table and no interpolation
+            can be done.
+        """
+        try:
+            return self._coeffs[imt]
+        except KeyError:
+            pass
```

Comment:
The purpose of ignoring the KeyError is unclear

Reply:
This is inherited from CoeffsTable: [This is inherited from CoeffsTable: https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/gsim/base.py#L830](https://github.com/gem/oq-engine/blob/master/openquake/hazardlib/gsim/base.py#L830)